src/__swaps__/screens/Swap/hooks/useSwapEstimatedGasLimit.ts

```diff
@@ -98,8 +84,18 @@ export function useSwapEstimatedGasLimit(
       cacheTime: 60 * 1000, // 1min
       notifyOnChangeProps: ['data', 'isFetching'],
       keepPreviousData: true,
-      placeholderData: gasUnits.basic_swap[chainId],
+      enabled: !!chainId && !!quote && !!assetToSell && assetToSell.chainId === chainId,
+      initialData,
+      initialDataUpdatedAt: 0,
       ...config,
     }
   );
+
+  // we keepPreviousData so we can return the previous gasLimit while fetching
+  // which is great when refetching for the same chainId, but we don't want to keep the previous data
```

Comment:
If I switch assets under the same chain id, I might go through a completely different route and the gas limit can be completely different, so this is not a good idea unless you take into account the whole route + chain id.
TLDR;  Only when this condition is true:
`input asset  === prev Input asset && output asset === prev output asset &&. chainId ==== prevChainId` you can return the previous value, otherwise you need to get a new one

Reply:
its returning the previous value just while refetching the new one, and we are displaying that it's loading